### ISSUE TITLE
[BUG] Adding timeline event via API results in HTTP/500 message; event are added despite error message

### ISSUE THREAD

**Issue by cudeso** [dfir-iris/iris-web#58]
**Describe the bug**
If you post an event to the timeline (/case/timeline/event/add) and set the **event_category to 0** there's an HTTP/500 error message.

The container log give this output

```
app_1       |   File "/opt/venv/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
app_1       |     cursor.execute(statement, parameters)
app_1       | sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) insert or update on table "case_events_category" violates foreign key constraint "case_events_category_category_id_fkey"
app_1       | DETAIL:  Key (category_id)=(0) is not present in table "event_category".
```

The event is added however to the database!

**To Reproduce**
Code to reproduce:

```
event_category = 0
iris_data=json.dumps({"event_color": default_event_colour, "event_title": event_title, "event_content": line, "event_raw": line, "event_source": label, "event_assets": [], "event_category": event_category, "event_date": event_date, "event_time": event_time, "event_tz": event_tz, "event_in_graph": True, "event_in_summary": True,  "event_tags": "timesketch",  "cid": cid })
result = requests.post("{}/case/timeline/event/add".format(iris_host), headers=iris_headers, data=iris_data, verify=iris_verify)
```

**Comment by whikernel** [comment]
Thanks for reporting ! 
This is because Category ID 0 doesn't exist. Every IDs start at 1. Also the event categories list can be fetch with `manage/event-categories/list`. 

However the server should check the ID before trying to commit the DB transaction. We'll fix that 👍

**Comment by whikernel** [comment]
@cudeso We actually alrteady fixed this in v1.3.0. Could you please try ? 
Also, the API changed, you can find the updated version of this endpoint here : https://dfir-iris.github.io/_static/iris_api_reference_v1.0.1.html#operation/post-case-timeline-add-delete-event_id---copy

**Comment by cudeso** [comment]
Indeed; no errors in v1.3.0! Thanks for fixing.
